packages/compass-aggregations/src/components/focus-mode/focus-mode-stage-preview.tsx

```diff
@@ -68,6 +83,59 @@ export const FocusModePreview = ({
   const docCount = documents?.length ?? 0;
   const docText = docCount === 1 ? 'document' : 'documents';
   const shouldShowCount = !isLoading && docCount > 0;
+
+  const isPipelineOptionsMenuVisible = documents && documents.length > 0;
+
+  let content = null;
+
+  if (stageOperator === '$out') {
+    content = (
+      <div className={centerStyles}>
+        <OutStagePreivew index={stageIndex} />
+      </div>
+    );
+  } else if (stageOperator === '$merge') {
+    content = (
+      <div className={centerStyles}>
+        <MergeStagePreivew index={stageIndex} />
+      </div>
+    );
+  } else if (isMissingAtlasOnlyStageSupport) {
+    content = (
+      <div className={centerStyles}>
+        <AtlasStagePreview stageOperator={stageOperator ?? ''} />
+      </div>
+    );
+  } else if (isLoading) {
+    content = (
+      <div className={centerStyles}>
+        <div className={loaderStyles}>
+          <SpinLoader title='Loading' />
+          Loading Preview Documents... 
+        </div>
+      </div>
+    );
+  } else if (documents && documents.length > 0) {
+    content = (
+      <DocumentListView
+        docs={documents}
+        copyToClipboard={(doc: HadronDocument) => {
```

Comment:
Can you double-check that this is necessary? I think the types in the compass-crud for this method are provided and so it's a bit weird that it's required here to explicitly specify them